runtime/lib/backend/openqasm/braket_local_qubit.toml

```diff
@@ -1,84 +1,82 @@
-schema = 1
-
-[device]
-name = "braket.aws.qubit"
-
-[operators]
-# Observables supported by the device
-observables = [
-	"PauliX",
-	"PauliY",
-	"PauliZ",
-	"Hadamard",
-	"Hermitian",
-	"Tensor",
-]
+schema = 2
 
 # The union of all gate types listed in this section must match what
 # the device considers "supported" through PennyLane's device API.
-[[operators.gates]]
-native = [
-        "ISWAP",
-        "PSWAP",
-	"Hadamard",
-        "PauliX",
-        "PauliY",
-        "PauliZ",
-        "S",
-        "T",
-        "CNOT",
-        "CZ",
-        "SWAP",
-        "PhaseShift",
-        "RX",
-        "RY",
-        "RZ",
-        "CSWAP",
-	"QubitUnitary",
-	"MS",
-	"CY",
-	"SX",
-	"AAMS",
-	"ECR",
-	"GPi",
-	"GPi2",
-	"CPhaseShift00",
-	"CPhaseShift01",
-	"CPhaseShift10",
-	"Identity",
-	"IsingXX",
-	"IsingXY",
-	"IsingYY",
-	"IsingZZ",
-	"Toffoli",
-	"ControlledPhaseShift",
-]
+[operators.gates.native]
+
+ISWAP = {}
+PSWAP = {}
+Hadamard = {}
+PauliX = {}
+PauliY = {}
+PauliZ = {}
+S = {}
+T = {}
+CNOT = {}
+CZ = {}
+SWAP = {}
+PhaseShift = {}
+RX = {}
+RY = {}
+RZ = {}
+CSWAP = {}
+QubitUnitary = {}
+MS = {}
+CY = {}
+SX = {}
+AAMS = {}
+ECR = {}
+GPi = {}
+GPi2 = {}
+CPhaseShift00 = {}
+CPhaseShift01 = {}
+CPhaseShift10 = {}
+Identity = {}
+IsingXX = {}
+IsingXY = {}
+IsingYY = {}
+IsingZZ = {}
+Toffoli = {}
+ControlledPhaseShift = {}
 
 # Operators that should be decomposed according to the algorithm used
 # by PennyLane's device API.
 # Optional, since gates not listed in this list will typically be decomposed by
 # default, but can be useful to express a deviation from this device's regular
 # strategy in PennyLane.
-decomp = []
+[operators.gates.decomp]
 
 # Gates which should be translated to QubitUnitary
-matrix = [
-]
+[operators.gates.matrix]
+
+[operators.observables]
+# Observables supported by the device
+PauliX = {}
+PauliY = {}
+PauliZ = {}
+Hadamard = {}
+Hermitian = {}
+# Tensor,
+Projector = {}
+Sprod = {}
+Hamiltonian = { condition = ['noshots'] }
```

Comment:
My previous comment would fit nicely here as well since Hamiltonian can then be configured as:
```suggestion
Hamiltonian = { finiteshots = False }
```

In general there a couple more questions to consider though:
- the flag names: for instance `analytic`/`finiteshots` is the terminology typically used by PennyLane, whereas "exactshots" is a bit of a strange term (what are exact shots?)
- the way flags are specified
  - the flag type:
    - I guess since we use a table with flags as keys, each flag also needs a value, which right now are Booleans
    - since a Boolean value admits three states (True, False, absent), we should define what "absent" means/define flag default values
    - an alternative could be to use the flags presence as a Boolean flag, so instead of using a table we could use an array of strings (e.g. `CNOT = [ "invertible", "controllable" ]` vs `CNOT = { invertible = True, controllable = True, differentiable = False}`)
  - the flag default
      - right now the default for gates is False, which makes sense since a gate with all False modifiers still has the meaning of being supported for basic execution
      - an observable or measurement process on the other hand would not make sense to list with all False flags, since `Expval = {}` = `Expval = {analytic = False, finiteshots = False}` could just be omitted altogether

Either way there is no right or wrong answers here, as long we have a good definition of the spec :)

Reply:
> Defining absent as False would work for invertible, controllable and differentiable, but for finiteshouts it should have the meaning "not applicable", right?

Oh I see, well I think I was assuming a predetermined set of allowable options (since we have to parse them to give them meaning allowing arbitrary options not predefined in the specification is not very useful). So for gates allowable options would be:
- invertible
- controllable
- differentiable

and for observables and measurements:
- finiteshots
- analytic

The condition framework is fine by me, which I interpret similar to allowlist model but with all True defaults when absent. Which model is more convenient (allowlist/blocklist, True/False defaults) really depends on the particular configuration scenario, so let's just go with this 👍 

Have you considered using values for the flags instead of keys, like `CNOT = [ "invertible", "controllable" ]`?

Reply:
> Have you considered using values for the flags instead of keys, like CNOT = [ "invertible", "controllable" ]?

I'm ok with both styles, the only complication I see (again for both) is that one probably better to have `True`/positive value by default - the majority of gates are invertible and controllable. Now, for the flag approach, if we make positive value the default, the flags would need to be something like `[ "noninvertible", "notcontrollable" ]`. It looks strange in my eyes. In this sense, the boolean flags with `True` defaults seem more conventional.

Just an idea: maybe we can use something like `Gate = { lacks = ["controllable"] }` for a non-controllable gate?

Reply:
I think everything is done here, closing the discussion